src/generators/typescript/presets/CommonPreset.ts

```diff
@@ -0,0 +1,145 @@
+  const properties = model.properties || {};
+  const propertyKeys = [...Object.entries(properties)];
+  const marshalProperties = propertyKeys.map(([prop, propModel]) => {
+    const camelCasedProp = FormatHelpers.toCamelCase(prop);
+    const propMarshalReference = `json += \`"${camelCasedProp}": $\{this.${camelCasedProp}.marshal()},\`;`;
+    const propMarshal = `json += \`"${camelCasedProp}": $\{JSON.stringify(this.${camelCasedProp})},\`;`;
+    const propMarshalCode = propModel.$ref !== undefined ? propMarshalReference : propMarshal;
+    return `if(this.${camelCasedProp} !== undefined) {
+  ${propMarshalCode} 
+}`;
+  }).join('\n');
+
+  let marshalPatternProperties = '';
+  if (model.patternProperties !== undefined) {
+    for (const [pattern, patternModel] of Object.entries(model.patternProperties)) {
+      let patternPropertyName = getUniquePropertyName(model, `${pattern}${DefaultPropertyNames.patternProperties}`);
+      patternPropertyName = renderer.nameProperty(patternPropertyName, patternModel);
+      marshalPatternProperties += `if(this.${patternPropertyName} !== undefined) { 
+  for (const [key, value] of this.${patternPropertyName}.entries()) {
+    //Only render pattern properties which are not already a property
+    if(Object.keys(this).includes(String(key))) continue;
+    ${patternModel.$ref !== undefined ? 'json += `"${key}": ${value.marshal()},`;' : 'json += `"${key}": ${JSON.stringify(value)},`;'}    
+  }
+}`;
+    }
+  }  
+
+  let marshalAdditionalProperties = '';
+  if (model.additionalProperties !== undefined) {
+    let additionalPropertyName = getUniquePropertyName(model, DefaultPropertyNames.additionalProperties);
+    additionalPropertyName = renderer.nameProperty(additionalPropertyName, model.additionalProperties);
+    marshalAdditionalProperties = `if(this.${additionalPropertyName} !== undefined) { 
+  for (const [key, value] of this.${additionalPropertyName}.entries()) {
+    //Only render additionalProperties which are not already a property
+    if(Object.keys(this).includes(String(key))) continue;
+    ${model.additionalProperties.$ref !== undefined ? 'json += `"${key}": ${value.marshal()},`;' : 'json += `"${key}": ${JSON.stringify(value)},`;'}    
+  }
+}`;
+  }
+  return `public marshal() : string {
+  let json = '{'
+${renderer.indent(marshalProperties)}
+
+${renderer.indent(marshalPatternProperties)}
+
+${renderer.indent(marshalAdditionalProperties)}
+
+  //Remove potential last comma 
+  return \`$\{json.charAt(json.length-1) === ',' ? json.slice(0, json.length-1) : json}}\`;
+}`;
+} 
+
+/**
+ * Render `unmarshal` function based on model
+ */
+function renderUnmarshal({ renderer, model }: {
+  renderer: TypeScriptRenderer,
+  model: CommonModel,
+}): string {
+  const properties = model.properties || {};
+  const propertyKeys = [...Object.entries(properties)];
+  const unmarshalProperties = propertyKeys.map(([prop, propModel]) => {
+    const camelCasedProp = FormatHelpers.toCamelCase(prop);
```

Comment:
You shouldn't use the `FormatHelpers.toCamelCase` function, because someone can pass another format casing in the `renderer.nameProperty` function - use this last.